advisor-ui/src/pages/ScheduleTool/ScheduleTool.jsx

```diff
@@ -0,0 +1,197 @@
+import * as React from "react";
+import { useEffect, useState } from "react";
+import { Link } from "react-router-dom";
+
+import "./ScheduleTool.css";
+import Constraints from "../../components/Constraints/Constraints";
+import ScheduleDetails from "../../components/ScheduleDetails/ScheduleDetails";
+
+export default function ScheduleTool() {
+  const [constraints, setConstraints] = useState([]);
+  const [schedule, setSchedule] = useState([]);
+
+  const containDuplicate = (constraints, newConstraint) => {
+    let flag = false;
+    constraints.forEach((constraint) => {
+      if (
+        constraint.value === newConstraint.value &&
+        constraint.type === newConstraint.type
+      ) {
+        flag = true;
+      }
+    });
+    return flag;
+  };
+
+  const addConstraint = (type, value) => {
+    const newConstraint = { type, value };
+    if (containDuplicate(constraints, newConstraint)) {
+      return false;
+    } else {
+      setConstraints((prevConstraints) => [
+        ...prevConstraints,
+        { type, value },
+      ]);
+      return true;
+    }
+  };
+
+  const generateNewSchedule = () => {
+    setSchedule(years);
+  };
+
+  const years = [
```

Comment:
Think it'd make this file a bit more readable if this was moved to another util or constants file, just because it takes up quite a few lines.